File: app/src/main/java/com/parabola/newtone/mvp/view/TimeToSleepInfoView.java
package com.parabola.newtone.mvp.view;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface TimeToSleepInfoView extends MvpView {
    void updateTimeToEndText(String timeToEndText);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void closeScreen();
}

File: app/src/main/java/com/parabola/newtone/mvp/view/TimeToSleepInfoView.kt
package com.parabola.newtone.mvp.view

import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle
import moxy.viewstate.strategy.alias.OneExecution

@AddToEndSingle
interface TimeToSleepInfoView : MvpView {
    fun updateTimeToEndText(timeToEndText: String)

    @OneExecution
    fun closeScreen()
}
